src/components/TextLabel.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { ReactNode } from 'react';
+import { Text, StyleSheet, TextStyle } from 'react-native';
+import { headerSize } from '@src/theme/Dimensions';
+interface CustomTextProps {
+  children: ReactNode;
+  variant?: 'header' | 'label' | 'inputLabel' | 'errorLabel' | 'xSmall';
```

Comment:
if possible create enum for variant as in the switch you are also checking the values.